use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -174,32 +266,82 @@ main(int argc, char *argv[])
 
             NumActiveClients++;
             NumActiveClientsHWM++;
-            printf("Server: Client ID=%d joined. # active clients=%d (HWM=%d)\n",
-                   req.clientId, NumActiveClients, NumActiveClientsHWM);
+            printf("Server: Client ID=%d joined. Clock ID=%d, # active clients=%d (HWM=%d)\n",
+                   req.clientId, clock_id, NumActiveClients, NumActiveClientsHWM);
             break;
 
           case REQ_MT_INCR:
+            if (clock_gettime(clock_id, &ts0)) {    // Timing begins
+                errExit("XXX-clock_gettime-ts0");
+            }
+
             clientp = &ActiveClients[req.client_idx];
             assert(clientp->clientId == req.clientId);
 
-            // Construct response message
-            resp.mtype = REQ_MT_INCR;
-            resp.clientId = req.clientId;   // For this client
+            resp.mtype = REQ_MT_INCR;               // Construct response message
+            resp.clientId = req.clientId;           // For this client
             resp.client_idx = req.client_idx;
 
-            // Implement and record increment.
-            clientp->last_mtype = REQ_MT_INCR;
-            resp.counter = ++clientp->client_ctr;
+            clientp->last_mtype = REQ_MT_INCR;      // Prepare to increment
+            resp.counter = ++clientp->client_ctr;   // Do Increment
+            clientp->num_ops++;                     // Track # of operations
+
+            if (clock_gettime(clock_id, &ts1)) {    // Timing ends
+                errExit("clock_gettime-ts1");
+            }
+
+            nsec0 = timespec_to_ns(&ts0);
+            nsec1 = timespec_to_ns(&ts1);
+            elapsed_ns = (nsec1 - nsec0);
+
 #if L3_ENABLED
-            l3_log_simple("Server msg: ClientID=%d, Increment=%" PRIu64, resp.clientId, resp.counter);
+            // Record time-consumed. (NOTE: See clarification below.)
+            l3_log_simple("Server msg: Increment: ClientID=%d, "
+                          "Thread-CPU-time=%" PRIu64 " ns.",
+                          resp.clientId, elapsed_ns);
 #endif // L3_ENABLED
+
+            // Reuse variable to reacquire current TS for accumulation.
+            // Cumulative time-consumed gives a read into impact of any
+            // logging call that might be done in diff micro-benchmarks.
+            if (clock_gettime(clock_id, &ts1)) {    // Cumulative timing ends
```

Comment:
This metric of cumulative time for the operation, which includes logging call on L299, will give us the overhead incurred by L3-logging.